test/form-state/e2e.spec.ts

```diff
@@ -0,0 +1,134 @@
+import type { BrowserContext, Page } from '@playwright/test'
+
+import { expect, test } from '@playwright/test'
+import { addBlock } from 'helpers/e2e/addBlock.js'
+import { trackNetworkRequests } from 'helpers/e2e/trackNetworkRequests.js'
+import * as path from 'path'
+import { fileURLToPath } from 'url'
+
+import {
+  ensureCompilationIsDone,
+  initPageConsoleErrorCatch,
+  saveDocAndAssert,
+  throttleTest,
+} from '../helpers.js'
+import { AdminUrlUtil } from '../helpers/adminUrlUtil.js'
+import { initPayloadE2ENoConfig } from '../helpers/initPayloadE2ENoConfig.js'
+import { TEST_TIMEOUT_LONG } from '../playwright.config.js'
+
+const filename = fileURLToPath(import.meta.url)
+const dirname = path.dirname(filename)
+
+const title = 'Title'
+let context: BrowserContext
+
+test.describe('Form State', () => {
+  let page: Page
+  let postsUrl: AdminUrlUtil
+
+  test.beforeAll(async ({ browser }, testInfo) => {
+    testInfo.setTimeout(TEST_TIMEOUT_LONG)
+
+    const { payload, serverURL } = await initPayloadE2ENoConfig({ dirname })
+    postsUrl = new AdminUrlUtil(serverURL, 'posts')
+
+    context = await browser.newContext()
+    page = await context.newPage()
+    initPageConsoleErrorCatch(page)
+    await ensureCompilationIsDone({ page, serverURL })
+  })
+
+  test('collection — should re-enable fields after save', async () => {
+    await page.goto(postsUrl.create)
+    await page.locator('#field-title').fill(title)
+    await saveDocAndAssert(page)
+    await expect(page.locator('#field-title')).toBeEnabled()
+  })
+
+  test('should thread proper event argument to validation functions', async () => {
+    await page.goto(postsUrl.create)
+    await page.locator('#field-title').fill(title)
+    await page.locator('#field-validateUsingEvent').fill('Not allowed')
+    await saveDocAndAssert(page, '#action-save', 'error')
+
+    expect(true).toBe(true)
```

Comment:
should not be needed if you are up to date with main

Reply:
Pretend like you didn't see that 🤪

Reply:
Fixed.